compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3733,8 +3752,26 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
                     break;
                 }
 
-                if (isAssignable(sourceMember, targetMember, unresolvedTypes)) {
-                    sourceTypeIsNotAssignableToAnyTargetType = false;
+                if (sourceMember.tag == TypeTags.ERROR) {
+                    if (targetUnionIsDistinctError) {
+                        sourceTypeIsNotAssignableToAnyTargetType = !isErrorTypeAssignable((BErrorType) sourceMember,
+                                (BErrorType) targetMember, unresolvedTypes);
+                    } else {
+                        if (targetMember.tag == TypeTags.ERROR && allTargetTypesErrors) {
```

Comment:
Do we need this extra level here. Can be just else if I think.

Reply:
Changed